### DISCUSSION TITLE
Underline and Strikeout barely visible - anything I can do?

### DISCUSSION THREAD

**Discussion by kuiperzone** [AvaloniaUI/Avalonia#17198] [Q&A]
Setting TextDecoration for Underline and StrikeOut/Strikethro renders decoration that is barely visible with a normal font size because it seems to render the decoration in a grayed color, rather than the foreground color. The StrikeOut line seems unjustly thin.

![image](https://github.com/user-attachments/assets/b495338e-fe41-48d2-ac72-005c3cf05d04)

Anything I can do to render the decor in the foreground color? This might pose an "accessibility issue".

**Comment by stevemonaco** [comment]
`TextDecoration`s have some degree of customization. eg.

```xml
<Window.Resources>
    <TextDecorationCollection x:Key="CustomStrikethrough">
        <TextDecoration
            Location="Strikethrough"
            Stroke="Orange"
            StrokeThickness="4" />
    </TextDecorationCollection>
</Window.Resources>

<StackPanel>
    <TextBlock Text="Testing a custom strikethrough" TextDecorations="{StaticResource CustomStrikethrough}" />
    <TextBlock Text="Testing a custom strikethrough" TextDecorations="{StaticResource CustomStrikethrough}" />
    <TextBlock Text="Testing a custom strikethrough" TextDecorations="{StaticResource CustomStrikethrough}" />
</StackPanel>
```

![image](https://github.com/user-attachments/assets/b95b2c92-e9e7-45a1-942d-d2e892931010)

However, @Gillibald may want to double check the defaults as your example text decorations are barely legible.

**Reply by kuiperzone** [reply]
Hey! Thank you. Need to do this at runtime rather than in XAML. But your comment pointed in the right direction, so think I might be able to figure it out from the TextDecoration class.

**Comment by Gillibald** [comment]
We are currently not aligning text decorations to the pixel grid. Default location comes from the font itself but we might want to adjust given defaults so they are readable all the time.

**Reply by stevemonaco** [reply]
It's not the location that hampers the readability of the example text in the opening comment. The line thickness is very thin. The color seems less bright, too, but maybe it's just antialiasing and not opacity or a darker shade.

**Reply by kuiperzone** [reply]
It could just be anti-aliasing. It is more visible with black on a white background. I may try to increase the pixel thickness just a tad.
